### PR TITLE
Provide environment variable alternatives to CLI arguments

### PR DESCRIPTION
**Pull request by Molter73**
## Description

This change makes it so environment variables can be used to configure all arguments that can be provided via CLI parameters. This should in turn make configuring collector in k8s easier and simplify the command
to be used in our dockerfiles.

In order to prevent as many breaking changes as possible, CLI arguments will take precedence over environment variables. For TLS certificates a variable was added for specifying a directory holding the ca.pem,
cert.pem and key.pem files, and three other variables were added to be able to override the location of each of the files or explicitly setting all three if needed.

New environment variables added:
- ROX_COLLECTOR_LOG_LEVEL: Sets the log level to be used.
- ROX_COLLECTOR_SCRAPE_INTERVAL: Set the scrapping interval in seconds.
- ROX_COLLECTOR_SCRAPE_DISABLED: Completely disable scrapping.
- ROX_COLLECTOR_TLS_CERTS: Path to a directory holding TLS certificates.
- ROX_COLLECTOR_TLS_CA: Path to the CA certificate to be used in TLS.
- ROX_COLLECTOR_TLS_CLIENT_CERT: Path to the client certificate to be used in TLS.
- ROX_COLLECTOR_TLS_CLIENT_KEY: Path to the client key to be used in TLS.

Existing dockerfile environment variables that used to be passed through CLI args that should still be compatible:
- COLLECTOR_CONFIG
- COLLECTION_METHOD
- GRPC_SERVER

## Checklist
- [x] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

CI should be enough.

### CODE REVIEW

File: collector/lib/CollectorConfig.cpp [hunk 1]
```diff
@@ -102,48 +121,60 @@ void CollectorConfig::InitCollectorConfig(CollectorArgs* args) {
   if (args) {
```
Comment (Stringy) [inline collector/lib/CollectorConfig.cpp:121]: Not something for this PR necessarily, but this init function is huuuuuuge. I wonder if we could break it up a little? wdyt?

File: collector/lib/EnvVar.h [hunk 2]
```diff
@@ -46,6 +48,54 @@ class EnvVar {
   mutable std::once_flag init_once_;
 };
 
+template <typename T, typename ParseT>
+class OptionalEnvVar {
```
Comment (Stringy) [inline collector/lib/EnvVar.h:52]: This is a pretty Rusty class if ever I've seen one, and it's great :smile: :crab:

**Review by Stringy — APPROVED** [review]
Couple of minor comments but nothing that should block this from merging, LGTM!

File: collector/lib/EnvVar.h [hunk 3]
```diff
@@ -46,6 +48,54 @@ class EnvVar {
...
+    return tryValue().has_value();
+  }
+
+  // Will throw an exception if val_ is nullopt, make sure to
+  // call hasValue and check that is not the case before hand.
+  const T& value() const {
+    return tryValue().value();
+  }
+
+  const T& valueOr(const T& alt) {
```
[the remark this answers is not in the document]
Comment (Molter73) [reply collector/lib/EnvVar.h:68]: Well that's what `EnvVar` would be for, wouldn't it? So if you want a default you use that class, if you want to know whether the env var was set you use this class.

I only added this method to simplify the code in [here](https://github.com/stackrox/collector/blob/62ff4bc65ee0ea7c831b2da7e85557de79e3ff4e/collector/lib/CollectorConfig.cpp#L195-L213) a bit, though I'm still not very happy with that block.

Maybe if we make the default value an optional we can find a way to unify both classes, I'll have to give it some thought.

File: collector/lib/CollectorConfig.cpp (hunk 1 above)
Comment (Molter73) [reply collector/lib/CollectorConfig.cpp:121]: Yeah, it's gotten way out of hand. We can probably figure some way to split sections of the configuration up and just have private/protected methods that handle those individually or something. I might give it a shot in a follow up.

File: collector/lib/EnvVar.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (Molter73) [reply collector/lib/EnvVar.h:40]: I added the constructor with the default value and then realized that by actually changing `val_` to be optional in the original `EnvVar` class and making a constructor that delegates to making the default value be `std::nullopt` we can pretty much unify both classes into one.

The main difference between the two implementations is that now is up to the user of the class to know if a default is provided or not, if it is, calling `value()` is perfectly safe, if it is not, they need to check if a value was retrieved from an environment value manually or call `valueOr()`. If we want to keep a clear distinction, I think we can probably do something fancy with inheritance, WDYT?

[the remark this answers is not in the document]
Comment (Stringy) [reply collector/lib/EnvVar.h:40]: I think this is a fair compromise and the caller should already know whether or not there's a default value provided, so I don't think we necessarily lose anything by combining the two classes in this way.

**Pull request merged**